### PR TITLE
Fix All Tests Under tests/unit

### PR DESCRIPTION
**Pull request by tmichaeldb**
## Description

This PR fixes all tests in the top level of the directory `tests/unit`.

This does **not** fix tests in `tests/unit/ml_handlers`. Another PR will fix those tests

## Type of change

(Please delete options that are not relevant)

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: `tests/unit`
 - [x]   Verification Steps: Run tests locally (e.g. `python -m pytest .\tests\unit\test_project_structure.py -s -vv`)

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/interfaces/file/file_controller.py [hunk 1]
```diff
@@ -104,7 +104,12 @@ def save_file(self, name, file_path, file_name=None):
             file_dir.mkdir(parents=True, exist_ok=True)
             source = file_dir.joinpath(file_name)
             # NOTE may be delay between db record exists and file is really in folder
-            shutil.move(file_path, str(source))
+            try:
+                shutil.move(file_path, str(source))
+            except PermissionError as e:
+                logger.warning('Permission error while moving file %s: %s', file_path, str(e))
```
Comment (StpMax) [inline mindsdb/interfaces/file/file_controller.py:110]: is it right not to raise an exception in this case?

**Comment by ea-rus** [comment]
@tmichaeldb do `test_last` and `test_last_in_job`  functions work with this PR?

**Comment by ea-rus** [comment]
@tmichaeldb, here I returned these tests: https://github.com/mindsdb/mindsdb/pull/8694
- test_last was adapted to use dummy_data 
- test_last_in_job is checking input to data handler (does dummy_data supports it?) so I keep mocking of data handler 

BTW, what is the reason to replace mock of datahandler with dummy_data?

File: mindsdb/interfaces/file/file_controller.py (hunk 1 above)
Comment (ea-rus) [reply mindsdb/interfaces/file/file_controller.py:110]: Could the cause of the problem be: the file description is still opened outside of this function?
Is [this function](https://github.com/mindsdb/mindsdb/blob/aea53cd15306a08a0e1a12a42051f2d18969cd5d/tests/unit/executor_test_base.py#L262) is previous in call stack?

File: tests/unit/test_project_structure.py [hunk 2]
```diff
@@ -943,56 +934,3 @@ def test_inactive_job(self, scheduler):
         ret = self.run_sql('select * from jobs_history')
         # no history
         assert len(ret) == 0
-
-    def test_conditional_job(self, scheduler):
```
Comment (ea-rus) [inline tests/unit/test_project_structure.py:947]: this is a fresh test, it should work. lets keep it

**Review by ea-rus — CHANGES_REQUESTED** [review]

**Review by ea-rus — APPROVED** [review]

**Pull request merged**